aten/src/ATen/native/UpSample.h

```diff
@@ -111,11 +111,22 @@ static inline void upsample_3d_shape_check(
   }
 }
 
+template <typename scalar_t>
+static inline scalar_t compute_scales_value(
+    const double scale,
+    int64_t input_size,
+    int64_t output_size) {
+      return (scale > 0.)
+          ? static_cast<scalar_t>(1. / static_cast<scalar_t>(scale))
```

Comment:
It may be better to do this division is double-precision since scale is coming in as a double and cast it to float just once. So something like:
`? static_cast<scalar_t>(1.0 / scale)`